### PR TITLE
DynamoDB CDC: Add demo to support reading DynamoDB change data capture

### PR DESCRIPTION
**Pull request by amotl**
## About
To support people using CrateDB together with infrastructure components managed on other people's computers more easily.

## Documentation
Preview: https://cratedb-toolkit--205.org.readthedocs.build/io/dynamodb/cdc.html

## Details
A little insight into the nitty gritty details. Currently, we are focusing on AWS Lambda to use as an on-demand compute environment, but the fundamental infrastructure is aiming to be generic enough to work on other kinds of compute environments as well.

### DynamoDB CDC
This makes a start by processing CDC events from DynamoDB, in order to re-materialize them into a CrateDB table.

### Generic Components
The subsystem bears a few generic components to build upon, still in its infancy, but also suitable for similar operations on other data sources.

- A convenient [Infrastructure as code (IaC)](https://en.wikipedia.org/wiki/Infrastructure_as_code) procedure to define data pipelines on [AWS](https://en.wikipedia.org/wiki/Amazon_Web_Services).
- Written in Python, using [AWS CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/Welcome.html) stack deployments. To learn what’s behind, see also [How CloudFormation works](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-overview.html).
- Code for running on [AWS Lambda](https://en.wikipedia.org/wiki/AWS_Lambda) is packaged into [OCI](https://en.wikipedia.org/wiki/Open_Container_Initiative) images, for efficient delta transfers, built-in versioning, and testing purposes.

### CODE REVIEW

File: doc/io/dynamodb/cdc.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+purposes, including a DynamoDB Table, connected to a Kinesis Stream.  
+
+For example, choose those two variants:
+
+- IaC driver: [dynamodb_kinesis_lambda_oci_cratedb.py]
+- Record processor: [kinesis_lambda.py]
+
+Putting them next to each other into a directory, and adjusting
+`LambdaPythonImage(entrypoint_file=...)` to point to the second,
+should be enough to get you started.
```
Comment (amotl) [inline doc/io/dynamodb/cdc.md:87]: Please move to the top.

File: doc/io/dynamodb/cdc.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  'DELETE FROM "demo-sink";'
+```
+
+Query documents in DynamoDB table.
+```shell
+aws dynamodb execute-statement --statement \
+  "SELECT * FROM \"demo-source\";"
+```
+
+Truncate MongoDB collection.
```
Comment (amotl) [inline doc/io/dynamodb/cdc.md:187]: That's clearly a copy/paste error.

File: doc/io/dynamodb/cdc.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+```shell
+aws dynamodb execute-statement --statement \
+  "DELETE FROM \"demo-source\" WHERE ${READING_WHERE};"
+```
```
Comment (amotl) [inline doc/io/dynamodb/cdc.md:191]: Does it even work?

**Comment by amotl** [comment]
We need to merge because shipping. Please leave your review in retrospective.

File: doc/io/dynamodb/cdc.md (hunk 3 above)
Comment (amotl) [reply doc/io/dynamodb/cdc.md]: There is no "TRUNCATE TABLE" for DynamoDB. Fixed with.

See also: https://stackoverflow.com/a/75936534.

**Review by surister — COMMENTED** [review]
LGTM

**Pull request merged**